src/main/java/seedu/address/logic/parser/gradetrackerparsers/AddAssignmentParser.java

```diff
@@ -16,29 +16,54 @@
 import seedu.address.logic.parser.ParserUtil;
 import seedu.address.logic.parser.Prefix;
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.module.ModuleName;
 import seedu.address.model.module.grade.Assignment;
+import seedu.address.model.module.grade.AssignmentName;
+import seedu.address.model.module.grade.AssignmentPercentage;
+import seedu.address.model.module.grade.AssignmentResult;
 
 public class AddAssignmentParser implements Parser<AddAssignmentCommand> {
     @Override
     public AddAssignmentCommand parse(String args) throws ParseException {
         requireNonNull(args);
         Assignment newAssignment;
-        String moduleName;
-        ArgumentTokenizer tokenizer = new ArgumentTokenizer(args, PREFIX_NAME, PREFIX_ADD_NEW_ASSIGNMENT,
-                PREFIX_PERCENTAGE_ASSIGNMENT, PREFIX_RESULT_ASSIGNMENT);
+        AssignmentName assignmentName;
+        AssignmentPercentage assignmentPercentage;
+        AssignmentResult assignmentResult;
+        ModuleName moduleName;
+        ArgumentTokenizer tokenizer = new ArgumentTokenizer(args, PREFIX_NAME, PREFIX_ASSIGNMENT_NAME,
+                PREFIX_ASSIGNMENT_PERCENTAGE, PREFIX_ASSIGNMENT_RESULT);
         ArgumentMultimap argMultimap = tokenizer.tokenize();
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADD_NEW_ASSIGNMENT,
-                PREFIX_PERCENTAGE_ASSIGNMENT, PREFIX_RESULT_ASSIGNMENT)
-                || !argMultimap.getPreamble().isEmpty()) {
+        if ((!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ASSIGNMENT_NAME,
+                PREFIX_ASSIGNMENT_PERCENTAGE, PREFIX_ASSIGNMENT_RESULT)
+                || !argMultimap.getPreamble().isEmpty()) && (argMultimap.getValue(PREFIX_ASSIGNMENT_NAME).isPresent()
```

Comment:
The arePrefixesPrensent() method actually checks if the respective Prefix values are present in the argument multimap already. There is no need to check here again. So you can remove the parts such as argMultimap.getValue(PREFIX_ASSIGNMENT_NAME).isPresent() in the if clause.